src/computation/logic/algorithm-utilities.ts

```diff
@@ -84,23 +88,16 @@ export function distributeSeats(
             };
             seatResult.partyResults.push(currentPartyResult);
 
-            if (
-                currentQuotient > currentWinner.quotient &&
-                !illegalPartyCodes.has(result.partyCode) &&
-                result.percentage > districtThreshold
-            ) {
-                currentWinner = currentPartyResult;
+            if (!illegalPartyCodes.has(result.partyCode) && result.percentage > districtThreshold) {
+                currentWinners = updateWinners(currentWinners, currentPartyResult);
             }
         }
-        seatsWon[currentWinner.partyCode] += 1;
-        currentSeatsWon[currentWinner.partyCode] += 1;
 
-        const updatedDenominator = getDenominator(algorithm, seatsWon[currentWinner.partyCode], firstDivisor);
-        const updatedQuotient = currentWinner.votes / updatedDenominator;
-        currentWinner.denominator = updatedDenominator;
-        currentWinner.quotient = updatedQuotient;
+        const winner = _.sample(currentWinners);
```

Comment:
Why is a random currentWinner being chosen here?

Reply:
Oh, nevermind, just a poorly chosen name for `currentWinners` -- they are per definition cases of equal votes (and quotient).

Reply:
Yes, the naming could be improved.